wdl/src/main/scala/wdl/types/WdlFlavoredWomType.scala

```diff
@@ -0,0 +1,57 @@
+package wdl.types
+
+import scala.collection.JavaConverters._
+import wdl.{WdlExpression, WdlSyntaxErrorFormatter}
+import wdl.AstTools.EnhancedAstNode
+import wom.core.WorkflowSource
+import wom.types.{WdlBooleanType, WdlFloatType, WdlIntegerType, WdlType}
+import wom.types.WdlType.parser
+import wom.values.{WdlBoolean, WdlFloat, WdlInteger, WdlValue}
+
+object WdlFlavoredWomType {
+  implicit class FromString(val wdlType: WdlType) extends AnyVal {
+    /**
+      * Converts WDL source into a WdlValue of this type, if possible.
+      *
+      * @param workflowSource source code representing the WdlValue
+      * @return The WdlValue
+      */
+    //TODO: return a Try ?
+    def fromWorkflowSource(workflowSource: WorkflowSource): WdlValue = {
+      wdlType match {
+        case WdlFloatType => WdlFloat(workflowSource.toDouble)
+        case WdlIntegerType => WdlInteger(workflowSource.toInt)
+        case WdlExpressionType => WdlExpression.fromString(workflowSource)
+        case WdlBooleanType => WdlBoolean(workflowSource.toBoolean)
+        case WdlNamespaceType => ??? // This is what the original code was doing and clearly this is right.
```

Comment:
I guess we could call `WdlNamespace.load` here even though I really doubt we'd ever end up in this case

Reply:
Pretty sure this is only ever in tests so I'm not sure whether we really need it... I'd be surprised if anything ever hit this case tbh